src/arrayMethodJoin.js

```diff
@@ -4,8 +4,37 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    const separatorToString = '' + separator;
+    let result = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      let char = this[i];
```

Comment:
could you please write code from 16 line to 22 line(including 30 line) shorter?

maybe you can add char to string if it isn't null or undefined(using !==).

Reply:
Thank you for advices!=)